src/main/java/io/aiven/elasticsearch/storage/security/RsaKeysReader.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2020 Aiven Oy
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.aiven.elasticsearch.storage.security;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.security.KeyFactory;
+import java.security.KeyPair;
+import java.security.NoSuchAlgorithmException;
+import java.security.PrivateKey;
+import java.security.PublicKey;
+import java.security.spec.InvalidKeySpecException;
+import java.security.spec.PKCS8EncodedKeySpec;
+import java.security.spec.X509EncodedKeySpec;
+
+import org.bouncycastle.util.io.pem.PemReader;
+
+class RsaKeysReader {
+
+    static KeyPair readRsaKeyPair(final InputStream publicKeyIn,
+                                  final InputStream privateKeyIn) {
+        final var publicKey = readPublicKey(publicKeyIn);
+        final var privateKey = readPrivateKey(privateKeyIn);
+        return new KeyPair(publicKey, privateKey);
+    }
+
+    private static PublicKey readPublicKey(final InputStream in) {
+        try {
+            final var pemContent = readPemContent(new InputStreamReader(in));
+            final var keySpec = new X509EncodedKeySpec(pemContent);
+            final var kf = KeyFactory.getInstance("RSA");
+            return kf.generatePublic(keySpec);
+        } catch (final NoSuchAlgorithmException | InvalidKeySpecException e) {
+            throw new IllegalArgumentException("Couldn't generate key pair", e);
```

Comment:
Maybe this exception should originate from `readRsaKeyPair` (and the same in the neighbor method)?